### PR TITLE
Feature/notifications inline

### PR DESCRIPTION
**Pull request by kostasdano**
## Description
This is an implementation of the issue #58 + a fix on the Button Component:

- Inline Notification is implemented, with all possible types and variations

- Added "cursor: pointer" to the Button Component

## Screenshots

- Inline Notifications  [(Abstract link)](https://app.abstract.com/projects/96260c1d-f169-4306-ab88-2472d8e457fd/branches/e8803d0d-0c59-4eeb-9d4e-1cae0df12a08/commits/latest/files/d7cb764d-aa66-4fd6-90dd-4804917a3ed3/layers/336D3D69-30DA-4DD5-B855-5F18BBCE962A?collectionId=69d4231d-574e-4e8c-89e8-70a68ba6edbb&collectionLayerId=1532083c-ec58-4e0c-a1af-c54419f0b0bc)
Untitled

- Button Update
Snap3
Snap4

## Test Plan

Test scenarios:
- Inline Notification renders correctly
- Inline Notification primaryCTA works properly when clicked
- Inline Notification closeCTA works properly when clicked

### CODE REVIEW

File: src/components/Notification/Notification.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# Inline Notification
+
+Inline Notification
+
+<Preview>
+  <Story name="Inline Notifiction" parameters={{ decorators: [withKnobs] }}>
+    <Stack>
+      <PresentComponent name="" width={768}>
+        <Notification
+          message={text('info', 'Informative Message')}
```
Comment (panvourtsis) [inline src/components/Notification/Notification.stories.mdx:32]: use the same name as prop e.g `message={text('message', 'Informative Message')}`

We will avoid confusions from devs

File: src/components/Notification/subcomponents/CompactNotification/CompactNotification.style.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+export const notificationsContainer = (
+  withFilling: boolean,
+  variant: CompactNotificationVariants,
+  type: NotificationTypes
+) => (theme: Theme): SerializedStyles => css`
+  display: flex;
+  justify-content: space-between;
+  overflow: hidden;
+  width: ${variant === 'inline' ? '100%' : rem(489)};
```
Comment (panvourtsis) [inline src/components/Notification/subcomponents/CompactNotification/CompactNotification.style.ts:24]: i dont think `rem(489)` is correct

File: src/components/Notification/subcomponents/CompactNotification/CompactNotification.style.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  height: ${rem(56)};
+  border-left: ${!withFilling ? typeToThemePalette(theme, type) : 'none'} 4px solid;
+  border: ${withFilling ? typeToThemePalette(theme, type) : 'none'} 1px solid;
+  background: ${withFilling
+    ? type === 'success'
```
Comment (panvourtsis) [inline src/components/Notification/subcomponents/CompactNotification/CompactNotification.style.ts:29]: these if elses could be simplified

File: src/components/Notification/Notification.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import * as React from 'react';
+import CompactNotification from './subcomponents/CompactNotification';
+
+export type NotificationTypes = 'success' | 'error' | 'info' | 'alert';
```
Comment (panvourtsis) [inline src/components/Notification/Notification.tsx:6]: `alert` could be `warning` as we use this text for the same purpose

File: src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  withFilling: boolean;
+  /** The primary call-to-action label of the Notification */
+  primaryCTALabel?: string;
+  /** The primary call-to-action of the Notification */
+  primaryCTA?: () => void;
+  /** The closing call-to-action of the Notification */
+  closeCTA?: () => void;
+};
+
+const typeToColor = (type: string): AcceptedColorComponentTypes =>
```
Comment (panvourtsis) [inline src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx:37]: with the change i suggested at `CompactNotificationVariants` to change alert with warning that is the same thing with different naming this is not needed anymore

File: src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      <div css={infoContainer()}>
+        {withIcon && (
+          <div css={iconContainer()}>
+            <Icon name={type} color={typeToColor(type)} />
+          </div>
+        )}
+        <div>{message}</div>
+      </div>
+      <div css={actionsContainer()}>
+        {primaryCTA && (
```
Comment (panvourtsis) [inline src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx:61]: please check for both callback and text in order to render this

**Comment by panvourtsis** [comment]
I would like to see an implementation on some testing for all the rendering and callback functionality of the new component

File: src/components/Notification/subcomponents/CompactNotification/CompactNotification.style.ts (hunk 2 above)
Comment (kostasdano) [reply src/components/Notification/subcomponents/CompactNotification/CompactNotification.style.ts]: This is something leftover while investigating the banner component.

File: src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx (hunk 5 above)
Comment (kostasdano) [reply src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx]: Icon Component's props "name" and "color" accept the same value for all NotificationTypes except "warning":
Icon:name expects "alert" while Icon:color expects "warning"
Therefore, this converter is needed (updated to colorToIconName)

Comment (panvourtsis) [reply src/components/Notification/subcomponents/CompactNotification/CompactNotification.tsx]: if you change `NotificationTypes` as i suggested you will see what i mean. The problem here is that you use different types and specifically the text `alert`. Alert in our system is called warning.

**Review by panvourtsis — APPROVED** [review]
LGTM 🎊

**Pull request merged**